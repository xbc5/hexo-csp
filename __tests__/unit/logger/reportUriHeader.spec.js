const { reportUriHeader: factory } = require("../../../lib/logger");

const fixture = (
  reportUrl = "https://my-logger.com",
  directives = "default-src https://directive1.com; img-src https://directive2.com"
) => {
  const setHeader = jest.fn();
  const next = jest.fn();
  const res = { setHeader };
  return {
    args: [{}, res, next],
    middleware: factory(reportUrl, directives),
    next,
    res,
  };
};

describe("when finished", () => {
  it("should call next", async () => {
    const { middleware, args, next } = fixture();
    middleware(...args);
    expect(next).toHaveBeenCalledTimes(1);
  });

  it("should not call next with any args (i.e. no error)", async () => {
    const { middleware, args, next } = fixture();
    middleware(...args);
    expect(next.mock.calls[0]).toMatchSnapshot();
  });
});

describe("when setting the Content-Security-Policy header", () => {
  it("should do it only once", async () => {
    const { middleware, args, res } = fixture();
    middleware(...args);
    expect(res.setHeader).toHaveBeenCalledTimes(1);
  });

  it("should set it with expected values", async () => {
    const { middleware, args, res } = fixture();
    middleware(...args);
    expect(res.setHeader.mock.calls[0]).toMatchSnapshot();
  });
});
