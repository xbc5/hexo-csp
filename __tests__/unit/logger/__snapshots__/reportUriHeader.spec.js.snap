// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when finished should not call next with any args (i.e. no error) 1`] = `Array []`;

exports[`when setting the Content-Security-Policy header should set it with expected values 1`] = `
Array [
  "Content-Security-Policy",
  "default-src https://directive1.com; img-src https://directive2.com; report-uri https://my-logger.com; report-to {\\"group\\":\\"csp-endpoint\\",\\"max_age\\":10886400,\\"endpoints\\":[{\\"url\\":\\"https://my-logger.com\\"}]};",
]
`;
